lib/executor/ramping_vus.go

```diff
@@ -493,154 +493,194 @@ var _ lib.Executor = &RampingVUs{}
 
 // Run constantly loops through as many iterations as possible on a variable
 // number of VUs for the specified stages.
-//
-// TODO: split up? since this does a ton of things, unfortunately I can't think
-// of a less complex way to implement it (besides the old "increment by 100ms
-// and see what happens)... :/ so maybe see how it can be split?
-// nolint:funlen,gocognit,cyclop
-func (vlv RampingVUs) Run(
-	parentCtx context.Context, out chan<- stats.SampleContainer, _ *metrics.BuiltinMetrics,
-) (err error) {
-	rawExecutionSteps := vlv.config.getRawExecutionSteps(vlv.executionState.ExecutionTuple, true)
-	regularDuration, isFinal := lib.GetEndOffset(rawExecutionSteps)
-	if !isFinal {
-		return fmt.Errorf("%s expected raw end offset at %s to be final", vlv.config.GetName(), regularDuration)
+func (vlv RampingVUs) Run(ctx context.Context, _ chan<- stats.SampleContainer, _ *metrics.BuiltinMetrics) error {
+	stepsRaw := vlv.config.getRawExecutionSteps(vlv.executionState.ExecutionTuple, true)
```

Comment:
nit: I am not against(or for) the shorter names, but `rawSteps` is more "grammatically" correct IMO

Reply:
Yeah, that was actually for kind of grouping steps into raw and gracefuls.

Reply:
I guess this works as well :shrug: . I will leave this unresolved for others to give their opinion, but I don't require the change

Reply:
@imiric @codebien what do you guys think?

Reply:
I thought the same as Mihail when I read it.

>Yeah, that was actually for kind of grouping steps into raw and gracefuls. I thought of steps as an etymologic/behavioral differentiator. Maybe like: nDone, nFailed.

This makes sense, but the problem I see with it: it is not obvious so you should add a comment for explaining it (and I think we shouldn't do it because the name should be self-explanatory as much as possible).  For this reason, in the end, I prefer the correct grammatically form.

Reply:
from the mentions @yorugac is missing

Reply:
Yes, I agree with Mihail and Ivan: we're probably used to `nDone` but `rawSteps` :confused: